Mathlib/Analysis/FunctionalSpaces/SobolevInequality.lean

```diff
@@ -294,6 +294,7 @@ theorem lintegral_pow_le_pow_lintegral_fderiv_aux
     {u : (ι → ℝ) → F} (hu : ContDiff ℝ 1 u)
     (h2u : HasCompactSupport u) :
     ∫⁻ x, (‖u x‖₊ : ℝ≥0∞) ^ p ≤ (∫⁻ x, ‖fderiv ℝ u x‖₊) ^ p := by
+  letI := Classical.decEq ι
```

Comment:
Would you like to also drop the `DecidableEq ι` assumption on line 57?

Reply:
The definition of `T` and some others need decidable equality.

Reply:
I guess this is taking advantage of the current behavior of `variable` which drops unnecessary variables after processing the declaration. Can you just shadow the variables instead?